### ISSUE TITLE
Bug in mintinstall: It kills itself by installing flatpak of Handbrake

### ISSUE THREAD

**Issue by andhotz** [linuxmint/mintinstall#263]
I jus installed a completely new Mint 19.2 on a completely new disk. After installation, I let itself update and upgrade. Then installed some apps and flatpaks using mintinstall. After closing mintinstall, it does not come up again anymore. This is what is shown when I start mintinstall in a terminal:

__MintInstall: Detected system architecture: 'x86_64'
Installer: User pkgcache is most recent, using it.
Traceback (most recent call last):
  File "/usr/lib/linuxmint/mintinstall/mintinstall.py", line 580, in do_activate
    self.on_installer_ready()
  File "/usr/lib/linuxmint/mintinstall/mintinstall.py", line 885, in on_installer_ready
    self.process_matching_packages()
  File "/usr/lib/linuxmint/mintinstall/mintinstall.py", line 1606, in process_matching_packages
    remote_info = self.installer.cache.flatpak_remote_infos[remote_name]
KeyError: 'fr.handbrake.ghb-origin'
Error in sys.excepthook:
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/apport_python_hook.py", line 145, in apport_excepthook
    os.O_WRONLY | os.O_CREAT | os.O_EXCL, 0o640), 'wb') as f:
FileNotFoundError: [Errno 2] Datei oder Verzeichnis nicht gefunden: '/var/crash/_usr_lib_linuxmint_mintinstall_mintinstall.py.1000.crash'

Original exception was:
Traceback (most recent call last):
  File "/usr/lib/linuxmint/mintinstall/mintinstall.py", line 580, in do_activate
    self.on_installer_ready()
  File "/usr/lib/linuxmint/mintinstall/mintinstall.py", line 885, in on_installer_ready
    self.process_matching_packages()
  File "/usr/lib/linuxmint/mintinstall/mintinstall.py", line 1606, in process_matching_packages
    remote_info = self.installer.cache.flatpak_remote_infos[remote_name]
KeyError: 'fr.handbrake.ghb-origin'

** (mintinstall.py:3285): WARNING **: 16:59:09.142: Found icon of unknown type, skipping it: unknown__

I installed these items using mintinstall just before the crash:

_Normal appliactions:
    mc
    gparted
    virtualbox
    Google Earth
    Kate
    Audacity
    K3b
    Minitube
    Sox
    Shutter
    Gwenview
    Filezilla
    youtube-dl
    chromium
    Ttf-mscorefonts-installer
Flatpaks
    HandBrake
    LibreOffice 6.3
    Calibre
    Remmina
    Avidemux_

So it looks like one of these installations harmed mintinstall. Accrding to gm10 on Linux Mint Forum the bug was triggered when the flatpak of Handbrake was installed. 

Maybe this does not add light to the problem, but I do not know: "sudo apt-get update" deliivers the following: 

_Ign:1 http://dl.google.com/linux/earth/deb stable InRelease
OK:2 http://archive.canonical.com/ubuntu bionic InRelease                                         
OK:3 http://security.ubuntu.com/ubuntu bionic-security InRelease                                  
OK:4 http://dl.google.com/linux/earth/deb stable Release                                             
OK:5 http://pkg.adfinis-sygroup.ch/ubuntu bionic InRelease                                           
Ign:6 http://mirror.wtnet.de/linuxmint tina InRelease                        
OK:7 http://pkg.adfinis-sygroup.ch/ubuntu bionic-updates InRelease           
OK:8 http://pkg.adfinis-sygroup.ch/ubuntu bionic-backports InRelease
OK:9 http://mirror.wtnet.de/linuxmint tina Release   
Paketlisten werden gelesen... Fertig                 
N: Das Laden der konfigurierten Datei »main/binary-i386/Packages« wird übersprungen, da das Depot »http://dl.google.com/linux/earth/deb stable InRelease« die Architektur »i386« nicht unterstützt_

For those of you who do not speak german: The error message at the bottom of the output of apt-get says that it would jump over the configured file "main/binary-i386/Packages", because »http://dl.google.com/linux/earth/deb stable InRelease« would not support the architecture »i386«

Strange enough: Google Earth was installed without any hickup and runs smoothly. I'm stuck

Regards Andreas

**Comment by DiegoBranco** [comment]
I was told on the forums to post this here:

Installed Citra Emulator using this command:

`flatpak install https://flatpak.citra-emu.org/citra-nightly.flatpakref`

The emulator itself worked fine, however, running the software manager (mintinstall) does not work. Synaptic does, however, as does apt-get in the command line.

Running mintinstall in the command line throws the following error:

```
mintinstall
MintInstall: Detected system architecture: 'x86_64'
Installer: User pkgcache is most recent, using it.
Traceback (most recent call last):
  File "/usr/lib/linuxmint/mintinstall/mintinstall.py", line 580, in do_activate
    self.on_installer_ready()
  File "/usr/lib/linuxmint/mintinstall/mintinstall.py", line 885, in on_installer_ready
    self.process_matching_packages()
  File "/usr/lib/linuxmint/mintinstall/mintinstall.py", line 1606, in process_matching_packages
    remote_info = self.installer.cache.flatpak_remote_infos[remote_name]
KeyError: 'org.citra.citra-nightly-origin'
Error in sys.excepthook:
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/apport_python_hook.py", line 145, in apport_excepthook
    os.O_WRONLY | os.O_CREAT | os.O_EXCL, 0o640), 'wb') as f:
FileNotFoundError: [Errno 2] No such file or directory: '/var/crash/_usr_lib_linuxmint_mintinstall_mintinstall.py.1000.crash'

Original exception was:
Traceback (most recent call last):
  File "/usr/lib/linuxmint/mintinstall/mintinstall.py", line 580, in do_activate
    self.on_installer_ready()
  File "/usr/lib/linuxmint/mintinstall/mintinstall.py", line 885, in on_installer_ready
    self.process_matching_packages()
  File "/usr/lib/linuxmint/mintinstall/mintinstall.py", line 1606, in process_matching_packages
    remote_info = self.installer.cache.flatpak_remote_infos[remote_name]
KeyError: 'org.citra.citra-nightly-origin'

** (mintinstall.py:2984): WARNING **: 18:51:13.552: Found icon of unknown type, skipping it: unknown

```
Uninstalling it makes it work again, but i would like to keep both programs. how do i fix this?

For further reference, here's the output of apt policy mintinstall:

`mintinstall:
  Instalados: 8.0.0
  Candidato:  8.0.0
  Tabla de versión:
 *** 8.0.0 500
        500 http://mintmirror.math.washington.edu/linuxmint-repo tina/main amd64 Packages
        500 http://mintmirror.math.washington.edu/linuxmint-repo tina/main i386 Packages
        100 /var/lib/dpkg/status`

Also tried Updating flatpak from their official PPA, but it does not help.

**Comment by mtwebster** [comment]
I can reproduce this, I'll try to fix asap.

**Comment by mtwebster** [comment]
If you'd like, you can install this package:

https://www.dropbox.com/s/gvk11yic5ukah3v/mint-common_2.0.7_all.deb?dl=0

Download it, open a terminal in that folder, and run `sudo dpkg -i mint-common*.deb`

I'm trying to get a point release as well, but this should fix the issue for now, please let me know if it doesn't.

**Comment by DiegoBranco** [comment]
Can confirm it does work, thank you!

**Comment by andhotz** [comment]
Works perfectly. Thank you very much.
Regards
Andreas
--
Andreas Hotz
Weidstrasse 12
6300 Zug

**Comment by mathieuaubert** [comment]
Hello
Sorry for my level in English.
I have the same problem.
I have Mint 19.3 Cinnamon, fresh install
And Mintinstall have a problem :
`MintInstall: Detected system architecture: 'i686'
Installer: User pkgcache is most recent, using it.
Traceback (most recent call last):
  File "/usr/lib/linuxmint/mintinstall/mintinstall.py", line 580, in do_activate
    self.on_installer_ready()
  File "/usr/lib/linuxmint/mintinstall/mintinstall.py", line 894, in on_installer_ready
    self.sync_installed_apps()
  File "/usr/lib/linuxmint/mintinstall/mintinstall.py", line 1156, in sync_installed_apps
    installed_packages = self.installer.cache.get_manually_installed_packages()
  File "/usr/lib/python3/dist-packages/mintcommon/installer/cache.py", line 307, in get_manually_installed_packages
    installed_packages_apt = self._get_manually_installed_debs()
  File "/usr/lib/python3/dist-packages/mintcommon/installer/cache.py", line 291, in _get_manually_installed_debs
    installer_log = gzip.open(installer_log, "r").read().decode('utf-8').splitlines()
  File "/usr/lib/python3.6/gzip.py", line 276, in read
    return self._buffer.read(size)
  File "/usr/lib/python3.6/gzip.py", line 482, in read
    raise EOFError("Compressed file ended before the "
EOFError: Compressed file ended before the end-of-stream marker was reached
Error in sys.excepthook:
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/apport_python_hook.py", line 145, in apport_excepthook
    os.O_WRONLY | os.O_CREAT | os.O_EXCL, 0o640), 'wb') as f:
FileNotFoundError: [Errno 2] Aucun fichier ou dossier de ce type: '/var/crash/_usr_lib_linuxmint_mintinstall_mintinstall.py.1000.crash'
`

I have test mint-common 2.0.7 (https://www.dropbox.com/s/gvk11yic5ukah3v/mint-common_2.0.7_all.deb?dl=0) and 2.0.9 but the problem persist

Thanks

**Comment by andhotz** [comment]
The problem as I posted it was solved and closed for mint 19.2. I switched to mint 19.3 recently and cannot confirm that the problem pops up again. mathieuaubert states he uses a fresh install, so I suggest to apply all updates and try again. If the problem persists, I cannot helpt further. However please note that this issue is closed, so I don't think anybody will notice the problem, or do they?

**Comment by mathieuaubert** [comment]
thx for you answer @andhotz , i have opened a new ticket : https://github.com/linuxmint/mintinstall/issues/271
